scripts/version/version.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+
+import argparse
+import subprocess
+
+
+def get_tag():
+    """Get the tag using git decsribe."""
+    git_describe = (
+        subprocess.check_output(
+            ["git", "describe", "--tags", "--always", "--long", "--dirty=+"]
+        )
+        .decode("utf-8")
+        .strip()
+    )
+
+    return git_describe
+
+
+def parse_tag(tag):
+    """Parse the tag into major, minor, patch, and tweak version numbers.
+
+    Args:
+        tag (str): The tag to parse.
+    """
+    tag = tag.replace("v", "")
+    tag = tag.replace("-", ".")
+    parts = tag.split(".")
+
+    if len(parts) >= 4:
+        version_major = parts[0]
+        version_minor = parts[1]
+        version_patch = parts[2]
+        version_tweak = parts[3]
+    else:
+        version_major = "0"
+        version_minor = "0"
+        version_patch = "0"
+        version_tweak = "0"
+
+    return version_major, version_minor, version_patch, version_tweak
+
+
+def main():
+    """Create a Zephyr compatible VERSION file."""
+    parser = argparse.ArgumentParser(
+        prog="version",
+        description="Generate a Zephyr compatible VERSION file from a provided tag or from git describe. If the tag is not valid (or does not exists), the version will be set to 0.0.0-0.",
+    )
+    parser.add_argument(
+        "-f",
+        "--file",
+        metavar="FILE",
+        type=str,
+        required=True,
+        help="The file path to write the version information to.",
+    )
+    parser.add_argument(
+        "-t",
+        "--tag",
+        metavar="TAG",
+        type=str,
+        required=False,
+        help='The tag to use for version information in the format "vMAJOR.MINOR.PATCH-TWEAK". If not provided, git decsribe will be used.',
```

Comment:
Nit: I would add the angled braces and example input to better present the required format (that leading "v" is a bit hidden).

```suggestion
        help='The tag to use for version information in the format "v<MAJOR>.<MINOR>.<PATCH>-<TWEAK>", eg. v1.2.3.-4. If not provided, git describe will be used.',
```